File: backend/app/schemas/estimate.py
# app/schemas/estimate.py
from typing import List, Optional
from pydantic import BaseModel, Field


class EstimateRequest(BaseModel):
    meal_text: str = Field(min_length=1, max_length=2000)


class BreakdownItem(BaseModel):
    item: str
    calories: float
    protein: float
    carbs: float
    fibre: float
    fats: float


class EstimateResponse(BaseModel):
    calories: float
    protein: float
    carbs: float
    fibre: float
    fats: float
    confidence: Optional[float] = 0.85
    note: Optional[str] = None
    breakdown: Optional[List[BreakdownItem]] = None

File: frontend/src/models/estimate.ts
// src/models/estimate.ts
export interface NutritionBreakdownItem {
  item: string;
  calories: number;
  protein: number;
  carbs: number;
  fibre: number;
  fats: number;
}

export interface EstimateResponse {
  calories: number;
  protein: number;
  carbs: number;
  fibre: number;
  fats: number;
  confidence?: number;
  note?: string;
  breakdown?: NutritionBreakdownItem[];
}

export interface EstimateRequest {
  meal_text: string;
}
